github/enterprise_manage_ghes.go

```diff
@@ -0,0 +1,163 @@
+type SystemRequirementsNodes struct {
+	Hostname    *string                               `json:"hostname"`
+	Status      *string                               `json:"status"`
+	RolesStatus []*SystemRequirementsNodesRolesStatus `json:"roles_status"`
+}
+
+// SystemRequirementsNodesRolesStatus represents the status of a role on a system node.
+type SystemRequirementsNodesRolesStatus struct {
+	Status *string `json:"status"`
+	Role   *string `json:"role"`
+}
+
+// NodeReleaseVersions represents a response from the GetReplicationStatus method.
+type NodeReleaseVersions struct {
+	Hostname *string          `json:"hostname"`
+	Version  *ReleaseVersions `json:"version"`
+}
+
+// ReleaseVersions holds the release version information of the node.
+type ReleaseVersions struct {
+	Version   *string `json:"version"`
+	Platform  *string `json:"platform"`
+	BuildID   *string `json:"build_id"`
+	BuildDate *string `json:"build_date"`
+}
+
+// CheckSystemRequirements checks if GHES system nodes meet the system requirements.
+//
+// GitHub API docs: https://docs.github.com/enterprise-server@3.15/rest/enterprise-admin/manage-ghes#get-the-system-requirement-check-results-for-configured-cluster-nodes
+//
+//meta:operation GET /manage/v1/checks/system-requirements
+func (s *EnterpriseService) CheckSystemRequirements(ctx context.Context) (*SystemRequirements, *Response, error) {
+	u := "manage/v1/checks/system-requirements"
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	systemRequirements := new(SystemRequirements)
+	resp, err := s.client.Do(ctx, req, systemRequirements)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return systemRequirements, resp, nil
+}
+
+// ClusterStatus gets the status of all services running on each cluster node.
+//
+// GitHub API docs: https://docs.github.com/enterprise-server@3.15/rest/enterprise-admin/manage-ghes#get-the-status-of-services-running-on-all-cluster-nodes
+//
+//meta:operation GET /manage/v1/cluster/status
+func (s *EnterpriseService) ClusterStatus(ctx context.Context) (*ClusterStatus, *Response, error) {
+	u := "manage/v1/cluster/status"
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	clusterStatus := new(ClusterStatus)
+	resp, err := s.client.Do(ctx, req, clusterStatus)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return clusterStatus, resp, nil
+}
+
+// ReplicationStatus gets the status of all services running on each replica node.
+//
+// GitHub API docs: https://docs.github.com/enterprise-server@3.15/rest/enterprise-admin/manage-ghes#get-the-status-of-services-running-on-all-replica-nodes
+//
+//meta:operation GET /manage/v1/replication/status
+func (s *EnterpriseService) ReplicationStatus(ctx context.Context, opts *NodeQueryOptions) (*ClusterStatus, *Response, error) {
+	u, err := addOptions("manage/v1/replication/status", opts)
+	if err != nil {
+		return nil, nil, err
```

Comment:
Please try to make sure that test coverage is not missing for simple things like this.
We have an established pattern in this repo to call test helper functions like `testBadOptions`
and `testNewRequestAndDoFailure`, etc.

Reply:
You are absolutely right that there are many endpoints with only `opts` that we cannot use `testBadOptions` with.  I apologize for highlighting this one, as it indeed is a case where `testBadOptions` won't help.  For this one, we can simply ignore the CodeCov warning.  Thanks.